pkg/discovery/discovery.go

```diff
@@ -9,6 +9,12 @@ import (
 	"github.com/docker/infrakit/pkg/plugin"
 )
 
+// Do this once, globally to ensure the discovery directory exists
+func init() {
```

Comment:
If you're importing a package that does discovery based on the presence of a directory, why is creating it a bad thing?